### PR TITLE
Solution id is string

### PR DESCRIPTION
**Pull request by sunce86**
# Description
This PR changes the type of the `id` column in the `proposed_solutions` table from `i64` to `numeric(78,0)`. This adjustment prevents potential overflow issues from `u64` values in the driver API, which could previously lead to [failed u64 -> i64 conversions](https://github.com/cowprotocol/services/blob/main/crates/autopilot/src/infra/persistence/mod.rs#L154) and errors when saving proposed solutions.

Alternative approaches considered:

1. Dropping the `id` column: While feasible, this would prevent reusing the existing `domain::competition::Solution` object in the autopilot and would require either creating a new domain object or generating IDs dynamically when reading the solution from database, both of which are not great ideas.
2. Changing the API type to `u32`: A more invasive change, requiring modifications to the driver API and adjustments by solvers (e.g., Copium, which sometimes reports IDs exceeding `i64::MAX`).

After evaluating these options, I chose the least disruptive solution of converting `id` to `numeric(78,0)`.

# Changes

- [ ] Change the type of "id" column, from BigInt to numeric(78,0)

## How to test
Existing tests.

### CODE REVIEW

File: database/README.md [hunk 1]
```diff
@@ -352,7 +352,7 @@ All solutions reported by solvers, that were part of a solver competition. A sol
 ---------------|-----------|----------|--------
  auction\_id   | bigint    | not null | auction for which the solution was proposed 
  uid           | bigint    | not null | unique id of the proposed solution within a single auction
- id            | bytea     | not null | id of the proposed solution as reported by the solver
```
Comment (squadgazzz) [inline database/README.md:355]: Initially, there was a typo, so the ID's type is `bigint,` right?

File: crates/database/src/solver_competition.rs [hunk 2]
```diff
@@ -140,7 +140,7 @@ async fn save_solutions(
     builder.push_values(solutions.iter(), |mut b, solution| {
         b.push_bind(auction_id)
             .push_bind(solution.uid)
-            .push_bind(solution.id)
+            .push_bind(solution.id.clone())
```
Comment (squadgazzz) [inline crates/database/src/solver_competition.rs:143]: Redundant cloning?
```suggestion
            .push_bind(&solution.id)
```

**Review by squadgazzz — COMMENTED** [review]
Was `NUMERIC` considered instead of `TEXT` to maintain the numeric ID structure?

**Review by MartinquaXD — COMMENTED** [review]
openapi.yaml files need to be updated. At least some part currently mention integers.

**Comment by sunce86** [comment]
> Was `NUMERIC` considered instead of `TEXT` to maintain the numeric ID structure?

You are right! I was a bit hasty here and did not think too well of the type. Switched to numeric that we use for U256 values.

**Comment by sunce86** [comment]
> openapi.yaml files need to be updated. At least some part currently mention integers.

We don't change the api type here.

**Review by MartinquaXD — APPROVED** [review]

**Comment by sunce86** [comment]
@squadgazzz merging to include into release. let me know if you notice anything else

**Pull request merged**